packages/@react-spectrum/utils/src/useHasChild.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {RefObject, useLayoutEffect, useState} from 'react';
+
+export function useHasChild(query: string, ref: RefObject<HTMLElement>) {
+  let [hasChild, setHasChild] = useState(true);
+  useLayoutEffect(() => {
+    setHasChild(!!(ref.current && ref.current.querySelector(query)));
```

Comment:
use a boolean so that if the child element changes, we don't cause any unnecessary rerenders